api/scenario_api_utils/include/scenario_api_utils/scenario_api_utils.h

```diff
@@ -46,28 +33,32 @@ struct Pose2D
 };
 
 template <class T>
-T waitForParam(const ros::NodeHandle & nh, const std::string & key)
+T waitForParam(const std::shared_ptr<rclcpp::Node> & node, const std::string & key)
 {
   T value;
-  ros::Rate rate(0.5);
-  while (ros::ok()) {
-    const auto result = nh.getParam(key, value);
-    if (result) {
-      return value;
+  // TODO: replace this with Rate that works based on ROS Clock when it is available
+  // https://github.com/ros2/rclcpp/pull/1373
+  rclcpp::WallRate rate(0.5);
+  while (!rclcpp::ok()) {
+    try{
+      const auto result = node->declare_parameter(key).get<T>();
+      return result;
+    } catch(...)
+    {
```

Comment:
<you can remove one set of braces

Reply:
Do you mean we can remove braces because it's a single-line?
I personally don't mind either way, but at least ROS guideline says that we should have braces even for single line.
https://index.ros.org/doc/ros2/Contributing/Code-Style-Language-Versions/#always-use-braces

Reply:
Not sure what I meant here. Looks fine as is